client/src/components/search/search.component.tsx

```diff
@@ -84,14 +86,8 @@ class Search extends React.PureComponent<RouteComponentProps<{}>> {
   }
   
   private onInputChanged (): void {
-    const params = {
-      query: this.inputElement.value
-    };
-    
-    if (!params.query) {
-      delete params.query;
-    }
-    
+    const query = this.inputElement.value;
+    const params = query ? ({query}) : ({});
     this.props.history.push(`/search?${queryString.stringify(params)}`);
```

Comment:
I will write somewhere codestyle, but I prefer to separate declarations from calls by blank line